includes/Classifai/Providers/Azure/Language.php

```diff
@@ -0,0 +1,391 @@
+	 */
+	public function request_summary( $endpoint_url, $api_key, $post_content, $post_id ) {
+		$endpoint_url = add_query_arg(
+			'api-version',
+			static::API_VERSION,
+			$endpoint_url . static::ANALYZE_TEXT_ENDPOINT
+		);
+
+		$body = [
+			'analysisInput' => [
+				'documents' => [
+					[
+						'id'       => '1',
+						'language' => 'en',
+						'text'     => $post_content,
+					],
+				],
+			],
+			'tasks'         => [
+				[
+					'kind'       => 'AbstractiveSummarization',
+					'taskName'   => 'Classifai Summarization ' . $post_id,
+					'parameters' => [
+						/**
+						 * Filter the summary length.
+						 * Possible values are 'oneSentence', 'short', 'medium', 'long'.
+						 * Default is 'oneSentence'.
+						 *
+						 * @since 3.2.0
+						 * @hook classifai_azure_language_summary_length
+						 * @param string $summary_length The summary length.
+						 * @return string
+						 */
+						'summaryLength' => apply_filters( 'classifai_azure_language_summary_length', 'oneSentence' ),
+					],
+				],
+			],
+		];
+
+		$request = wp_remote_post(
+			$endpoint_url,
+			[
+				'headers' => [
+					'Ocp-Apim-Subscription-Key' => $api_key,
+					'Content-Type'              => 'application/json',
+				],
+				'body'    => wp_json_encode( $body ),
+			]
+		);
+
+		$headers = wp_remote_retrieve_headers( $request );
+
+		if ( ! is_wp_error( $request ) ) {
+			$response = json_decode( wp_remote_retrieve_body( $request ) );
+			if ( ! empty( $response->error ) ) {
+				return new WP_Error( 'summary_request', $response->error->message );
+			} elseif ( empty( $headers['operation-location'] ) ) {
+				return new WP_Error( 'summary_request', esc_html__( 'There was an error requesting the summary.', 'classifai' ) );
+			} else {
+				return $headers['operation-location'];
+			}
+		}
+
+		return $request;
+	}
+
+	/**
+	 * Get the text analysis job response
+	 *
+	 * The endpoint URL is returned in the 'operation-location' header of the initial request.
+	 * The job response will return a 'status' property if the job is completed.
+	 * If the job is not completed, wait a second and try again.
+	 *
+	 * @param string $url The URL to analyze.
+	 * @see https://learn.microsoft.com/en-us/azure/ai-services/language-service/summarization/quickstart?tabs=text-summarization%2Cmacos&pivots=rest-api
+	 * @return mixed
+	 */
+	private function retrieve_summary( $url ) {
+		$api_key = $this->feature_instance->get_settings( static::ID )['api_key'];
+
+		$request = safe_wp_remote_get(
+			$url,
+			[
+				'headers' => [
+					'Ocp-Apim-Subscription-Key' => $api_key,
+					'Content-Type'              => 'application/json',
+				],
+			]
+		);
+
+		$summary = '';
+
+		if ( ! is_wp_error( $request ) ) {
+			$response = json_decode( wp_remote_retrieve_body( $request ) );
+			if ( ! empty( $response->error ) ) {
+				return new WP_Error( 'auth', $response->error->message );
+			}
+
+			while ( 'succeeded' !== $response->status ) {
+				sleep( .5 );
```

Comment:
A downside of this approach is that if you navigate away from the page, the process will quit.
We can use Action Scheduler for background processing which we're using for the Classification feature. (#779)

@dkotter do you have any thoughts on this?

Reply:
I think the problem here just in quickly looking at the code is that this method gets called from a REST request. The easy fix is what we're doing here, just pausing and retrying until we get a successful response.

I do wonder what happens if the response never succeeds? Would this code just be stuck in this loop?

Action Scheduler would be a nicer fix here but would be more complicated with how everything else is set up in ClassifAI. Basically when the `Generate Excerpt` button is clicked, the initial request would trigger Action Scheduler and then on the javascript side, we would need to continue polling until the request is done (and show the loading indicator the entire time).

That's doable but doesn't match how we handle things elsewhere so would need to think through how best to achieve that, especially in a way that would work for other Features and Providers in the future